### ISSUE TITLE
Compile warnings on OS X 10.10.5 with Xcode 7.2

### ISSUE THREAD

**Issue by imikejackson** [BlueQuartzSoftware/ITKImageProcessing#11]
I am getting a hundreds of warnings when compiling the ItkImageProcessing plugin. Here is a short snippet from the compiler output:

```
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:47:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:21:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkImportDream3DImageContainer.h:81:23: warning: 'AllocateElements' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    virtual Element * AllocateElements( ElementIdentifier size, bool UseDefaultConstructor = false ) const;
                      ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:155:22: note: in instantiation of template class 'itk::ImportDream3DImageContainer<unsigned long, float>' requested here
    typedef typename PixelContainerType::Pointer                               PixelContainerPointer;
                     ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:126:5: note: in instantiation of template class 'itk::Dream3DImage<float, 3>' requested here
    ImageType::PointType origin;
    ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkImportImageContainer.h:152:22: note: overridden virtual function is here
  virtual TElement * AllocateElements(ElementIdentifier size, bool UseDefaultConstructor = false) const;
                     ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:47:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:21:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkImportDream3DImageContainer.h:83:18: warning: 'DeallocateManagedMemory' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    virtual void DeallocateManagedMemory();
                 ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkImportImageContainer.h:154:16: note: overridden virtual function is here
  virtual void DeallocateManagedMemory();
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:47:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:206:22: warning: 'GetBufferPointer' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    virtual TPixel * GetBufferPointer()
                     ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:126:5: note: in instantiation of template class 'itk::Dream3DImage<float, 3>' requested here
    ImageType::PointType origin;
    ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkImage.h:226:20: note: overridden virtual function is here
  virtual TPixel * GetBufferPointer()
                   ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:47:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:210:28: warning: 'GetBufferPointer' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    virtual const TPixel * GetBufferPointer() const
                           ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkImage.h:228:26: note: overridden virtual function is here
  virtual const TPixel * GetBufferPointer() const
                         ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:48:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkInPlaceImageToDream3DDataFilter.h:28:18: warning: 'itk::InPlaceImageToDream3DDataFilter<float, 3>::SetInput' hides overloaded virtual function [-Woverloaded-virtual]
    virtual void SetInput(const ImageType *image);
                 ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:141:5: note: in instantiation of template class 'itk::InPlaceImageToDream3DDataFilter<float, 3>' requested here
    InPlaceImageToDream3DDataFilterType::Pointer filter = InPlaceImageToDream3DDataFilterType::New();
    ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkProcessObject.h:522:16: note: hidden overloaded virtual function 'itk::ProcessObject::SetInput' declared here: different number of parameters (2 vs 1)
  virtual void SetInput(const DataObjectIdentifierType & key, DataObject *input);
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:47:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:21:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkImportDream3DImageContainer.h:81:23: warning: 'AllocateElements' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    virtual Element * AllocateElements( ElementIdentifier size, bool UseDefaultConstructor = false ) const;
                      ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:155:22: note: in instantiation of template class 'itk::ImportDream3DImageContainer<unsigned long, int>' requested here
    typedef typename PixelContainerType::Pointer                               PixelContainerPointer;
                     ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:231:7: note: in instantiation of template class 'itk::Dream3DImage<int, 3>' requested here
      ImageType::PointType origin;
      ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkImportImageContainer.h:152:22: note: overridden virtual function is here
  virtual TElement * AllocateElements(ElementIdentifier size, bool UseDefaultConstructor = false) const;
                     ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:47:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:21:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkImportDream3DImageContainer.h:83:18: warning: 'DeallocateManagedMemory' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    virtual void DeallocateManagedMemory();
                 ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkImportImageContainer.h:154:16: note: overridden virtual function is here
  virtual void DeallocateManagedMemory();
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:47:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:206:22: warning: 'GetBufferPointer' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    virtual TPixel * GetBufferPointer()
                     ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:231:7: note: in instantiation of template class 'itk::Dream3DImage<int, 3>' requested here
      ImageType::PointType origin;
      ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkImage.h:226:20: note: overridden virtual function is here
  virtual TPixel * GetBufferPointer()
                   ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:47:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkDream3DImage.h:210:28: warning: 'GetBufferPointer' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
    virtual const TPixel * GetBufferPointer() const
                           ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkImage.h:228:26: note: overridden virtual function is here
  virtual const TPixel * GetBufferPointer() const
                         ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:48:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkInPlaceImageToDream3DDataFilter.h:28:18: warning: 'itk::InPlaceImageToDream3DDataFilter<int, 3>::SetInput' hides overloaded virtual function [-Woverloaded-virtual]
    virtual void SetInput(const ImageType *image);
                 ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:243:7: note: in instantiation of template class 'itk::InPlaceImageToDream3DDataFilter<int, 3>' requested here
      InPlaceImageToDream3DDataFilterType::Pointer filter = InPlaceImageToDream3DDataFilterType::New();
      ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkProcessObject.h:522:16: note: hidden overloaded virtual function 'itk::ProcessObject::SetInput' declared here: different number of parameters (2 vs 1)
  virtual void SetInput(const DataObjectIdentifierType & key, DataObject *input);
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:49:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkInPlaceDream3DDataToImageFilter.h:27:18: warning: 'itk::InPlaceDream3DDataToImageFilter<float, 3>::SetInput' hides overloaded virtual function [-Woverloaded-virtual]
    virtual void SetInput(DataContainer::Pointer &dc);
                 ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:307:5: note: in instantiation of template class 'itk::InPlaceDream3DDataToImageFilter<float, 3>' requested here
    FilterType::Pointer filter = FilterType::New();
    ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkProcessObject.h:522:16: note: hidden overloaded virtual function 'itk::ProcessObject::SetInput' declared here: different number of parameters (2 vs 1)
  virtual void SetInput(const DataObjectIdentifierType & key, DataObject *input);
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:12:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:49:
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/ITKImageProcessingFilters/itkInPlaceDream3DDataToImageFilter.h:27:18: warning: 'itk::InPlaceDream3DDataToImageFilter<int, 3>::SetInput' hides overloaded virtual function [-Woverloaded-virtual]
    virtual void SetInput(DataContainer::Pointer &dc);
                 ^
/Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingFilterTest.cpp:391:7: note: in instantiation of template class 'itk::InPlaceDream3DDataToImageFilter<int, 3>' requested here
      FilterType::Pointer filter = FilterType::New();
      ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Core/Common/include/itkProcessObject.h:522:16: note: hidden overloaded virtual function 'itk::ProcessObject::SetInput' declared here: different number of parameters (2 vs 1)
  virtual void SetInput(const DataObjectIdentifierType & key, DataObject *input);
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:13:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingReaderTest.cpp:56:
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Remote/SCIFIO/include/itkSCIFIOImageIO.h:94:16: warning: 'SupportsDimension' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  virtual bool SupportsDimension( unsigned long dim );
               ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/IO/ImageBase/include/itkImageIOBase.h:364:16: note: overridden virtual function is here
  virtual bool SupportsDimension(unsigned long dim)
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:13:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingReaderTest.cpp:56:
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Remote/SCIFIO/include/itkSCIFIOImageIO.h:98:16: warning: 'CanReadFile' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  virtual bool CanReadFile(const char* FileNameToRead);
               ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/IO/ImageBase/include/itkImageIOBase.h:313:16: note: overridden virtual function is here
  virtual bool CanReadFile(const char *) = 0;
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:13:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingReaderTest.cpp:56:
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Remote/SCIFIO/include/itkSCIFIOImageIO.h:107:16: warning: 'ReadImageInformation' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  virtual void ReadImageInformation();
               ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/IO/ImageBase/include/itkImageIOBase.h:326:16: note: overridden virtual function is here
  virtual void ReadImageInformation() = 0;
               ^
In file included from Plugins/ITKImageProcessing/ITKImageProcessingUnitTest.cpp:13:
In file included from /Users/mjackson/Workspace/DREAM3D_Plugins/ITKImageProcessing/Test/ITKImageProcessingReaderTest.cpp:56:
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modules/Remote/SCIFIO/include/itkSCIFIOImageIO.h:110:16: warning: 'Read' overrides a member function but is not marked 'override' [-Winconsistent-missing-override]
  virtual void Read(void* buffer);
               ^
/Users/Shared/DREAM3D_SDK/InsightToolkit-4.9.1/Modu
```

**Comment by fbudin69500** [comment]
@imikejackson : What version of Clang is in Xcode 7.2? I don't have a make available right now, but I could install the same version of your compiler in the docker image we use to compile and test DREAM3D to find and remove these warning.

**Comment by imikejackson** [comment]
508:[mjackson@cayde:Debug]$ clang --version
Apple LLVM version 7.0.2 (clang-700.1.81)
Target: x86_64-apple-darwin14.5.0
Thread model: posix

After some googling I get LLVM 3.7.0

**Comment by thewtex** [comment]
These warnings happen with newer clang with a C++11 build. To address them, the `ITK_OVERRIDE` macro is added after methods that override virtual methods from their base class.

**Comment by fbudin69500** [comment]
@thewtex Do you know if the warnings are visible with clang 3.5 or if if is only with versions that are more recent (like 3.7)?

**Comment by thewtex** [comment]
@fbudin69500 I am not sure what version they show up.

**Comment by thewtex** [comment]
This was addressed by #13 ?

**Comment by fbudin69500** [comment]
@thewtex yes.
